### ISSUE TITLE
Investigate the best way to load data.json

### ISSUE THREAD

**Issue by steveklabnik** [steveklabnik/rustdoc#76]
So, in order to preserve the ability to load off of the file system, we currently make a relative call to load data.json https://github.com/steveklabnik/rustdoc/blob/master/frontend/app/routes/application.js#L6

But, this means that if yo uhit any URL other than the root, it mis-loads it.

We should find some way to fix this. Loading with `/` works in every case *except* when you're browsing as `file://`. ☹️

**Comment by revin** [comment]
Hey @steveklabnik, so I'm coming into this completely out of the blue and looking for frontend-tagged issues and don't really have any Ember experience 😛 , but if the only issue is literally `file://` itself and we're always in a browser, would it be sufficient to do something like `fetch((document.location.protocol === 'file:' ? '/' : '') + 'data.json')` (possibly exploded into multiple lines for readability)?

**Comment by steveklabnik** [comment]
that's an interesting idea! Quite possibly!

**Comment by euclio** [comment]
Disclaimer: New to ember.

I'm trying to fix this, but I'm finding a bunch of questions on StackOverflow that say that you just *can't* run ember without a server, because the relative paths won't resolve correctly. In addition, even if you fix them, the router will break. There is an ember plugin that lets you render everything statically, but that would require us knowing all the visitable routes upfront.

Not sure if there's a solution here other than running the ember server, or booting out ember and keeping everything static.

**Comment by steveklabnik** [comment]
> that you just can't run ember without a server, because the relative paths won't resolve correctly.

Yeah so I've seen some buggy behavior when not loading up the root, even though by loading up the root it does work.

**Comment by mgattozzi** [comment]
I think [this stackoverflow](https://stackoverflow.com/questions/30183620/how-to-use-ember-js-without-server) post gives some insight as to why.

**Comment by steveklabnik** [comment]
Soooo I've been thinking about this, and I'm not even sure we need to load it as ajax. Consider current rustdoc, for example: the search index is a js file that defines an object, like this: https://doc.rust-lang.org/stable/search-index.js

we could do the same, and then not even make the ajax call at all, and this would remove this path issue.

**Comment by mgattozzi** [comment]
That seems reasonable. If we don't need to make any Ajax calls at all that would be great. The docs are best treated as a static site for the most part.

**Comment by euclio** [comment]
Won't we still run into the same issues with ember? It would still be
looking for the file at `/`.

**Comment by steveklabnik** [comment]
@euclio no, because the js gets concatenated, there's only one file.

**Comment by steveklabnik** [comment]
Oh well, that'd be at build time, it'd end up being two files. Hm.

**Comment by euclio** [comment]
Is this all going to live in the same route? If not, we would still need ember to resolve the paths correctly. If so, why not just generate static HTML from data.json?
